framework/validators/UniqueValidator.php

```diff
@@ -105,38 +105,39 @@ public function init()
 
     /**
      * @inheritdoc
+     * @since 2.0.11
      */
     public function validateAttribute($model, $attribute)
     {
         /* @var $targetClass ActiveRecordInterface */
         $targetClass = $this->targetClass === null ? get_class($model) : $this->targetClass;
         $targetAttribute = $this->targetAttribute === null ? $attribute : $this->targetAttribute;
-
-        if (is_array($targetAttribute)) {
-            $params = [];
-            foreach ($targetAttribute as $k => $v) {
-                $params[$v] = is_int($k) ? $model->$v : $model->$k;
-            }
-        } else {
-            $params = [$targetAttribute => $model->$attribute];
-        }
+        $params = $this->prepareParams($targetAttribute, $model, $attribute);
 
         foreach ($params as $value) {
             if (is_array($value)) {
                 $this->addError($model, $attribute, Yii::t('yii', '{attribute} is invalid.'));
-
                 return;
             }
         }
 
-        $query = $targetClass::find();
-        $query->andWhere($params);
+        $exists = $this->exists($targetClass, $params, $model);
 
-        if ($this->filter instanceof \Closure) {
-            call_user_func($this->filter, $query);
-        } elseif ($this->filter !== null) {
-            $query->andWhere($this->filter);
+        if ($exists) {
+            if (count($targetAttribute) > 1) {
+                $this->addComboNotUniqueError($model, $attribute);
+            } else {
+                $this->addError($model, $attribute, $this->message);
+            }
         }
+    }
+
+    /**
+     * Checks duplicates of the current model in the database
+     */
+    private function exists($targetClass, $params, $model)
```

Comment:
could be useful to have a naming more unique to prevent collisions in the ancestor tree. A private method "Exists" is risky in general.